src/Internal/functions.php

```diff
@@ -43,7 +43,7 @@ function parseUri(string $uri): array {
         );
     }
 
-    if (empty($host) || empty($port)) {
+    if ($host == "" || $port == 0) {
```

Comment:
`===` should be used instead of `==`.

Reply:
A host is always a string and a port is always an int, so there is no need for strict comparison

Reply:
You mean there's no need for a weak comparison? ;-) `==` is horrible and should be avoided where possible.